src/sinol_make/commands/run/__init__.py

```diff
@@ -360,78 +367,169 @@ def print_view(output_file=None):
+				if field == '' and self.args.programs == None and "sinol_expected_scores" in self.config:
+					for solution in change:
+						removed_solutions.add(solution[0])
+				# Only if sinol_make was run on all groups we should check if any of them were removed
+				elif field[1] == "expected" and self.args.tests == None and "sinol_expected_scores" in self.config:
+					for group in change:
+						removed_groups.add(group[0])
+			elif type == "change":
+				if field[1] == "expected": # Result for group changed
+					solution = field[0]
+					group = field[2]
+					old_result = change[0]
+					result = change[1]
+
+					print(util.warning("Solution %s passed group %d with status %s while it should pass with status %s." %
+										(solution, group, result, old_result)))
+
+		if len(added_solutions) > 0:
+			print(util.warning("Solutions were added: "), end='')
+			print(util.warning(", ".join(added_solutions)))
+		if len(removed_solutions) > 0:
+			print(util.warning("Solutions were removed: "), end='')
+			print(util.warning(", ".join(removed_solutions)))
+
+		if len(added_groups) > 0:
+			print(util.warning("Groups were added: "), end='')
+			print(util.warning(", ".join([str(group) for group in added_groups])))
+		if len(removed_groups) > 0:
+			print(util.warning("Groups were removed: "), end='')
+			print(util.warning(", ".join([str(group) for group in removed_groups])))
+
+		if expected_scores == new_expected_scores and \
+			len(added_solutions) == 0 and len(removed_solutions) == 0 and \
+			len(added_groups) == 0 and len(removed_groups) == 0:
+			print(util.info("Expected scores are correct!"))
+		else:
+			if self.args.apply_suggestions:
+				if "sinol_expected_scores" in self.config:
+					expected_scores = self.config["sinol_expected_scores"]
+				else:
+					expected_scores = {}
+
+				for solution in removed_solutions:
+					del expected_scores[solution]
+
+				for solution in expected_scores:
+					for group in removed_groups:
+						if group in expected_scores[solution]["expected"]:
+							del expected_scores[solution]["expected"][group]
+					expected_scores[solution]["points"] = self.calculate_points(expected_scores[solution]["expected"])
+
+				for solution in new_expected_scores.keys():
+					if solution in expected_scores:
+						for group, result in new_expected_scores[solution]["expected"].items():
+							expected_scores[solution]["expected"][group] = result
+						expected_scores[solution]["points"] = self.calculate_points(expected_scores[solution]["expected"])
+					else:
+						expected_scores[solution] = new_expected_scores[solution]
+
+
+				self.config["sinol_expected_scores"] = expected_scores
+				with open(os.path.join(os.getcwd(), "config.yml"), "w") as f:
+					yaml.dump(self.config, f, default_flow_style=None)
+				print(util.info("Saved suggested expected scores description."))
+			else:
+				util.exit_with_error("Use flag --apply_suggestions to apply suggestions.")
```

Comment:
We could certainly split the `validate_expected_scores` function into more smaller functions. I was planning to split the functions into different files in another PR after this one is merged.
I don't think that the `writer` is needed, because `pytest` has an easy way to capture printed text.

Reply:
Sounds good!